### DISCUSSION TITLE
Connection with Python MIP

### DISCUSSION THREAD

**Discussion by erelsgl** [cvxpy/cvxpy#1663] [Q&A]
Currently, if I want to use CVXPY with a high-performance MIP solver, such as CBC, I have to manually install CBC, which is quite uncomfortable, particularly when building libraries that should be included in other projects (on Windows, it can be installed using the cylp package, but on Ubuntu and Mac, the CBC binaries must be installed manually before cylp).

The [Python MIP package](https://www.python-mip.com/), according to their documentation, comes pre-installed with CBC binaries for most used operational systems (Windows, MacOS and Linux). So, if MIP is installed, theoretically CVXPY should be able to find CBC and use it. 

But currently, when I install mip and then cvxpy, it still says that CBC is not installed. 

Is there a way to make CVXPY find the CBC binaries that are installed with MIP? This will make it much easier to solve MIPs in CVXPY.

**Comment by SteveDiamond** [comment]
Well for your approach to work there would need to be a cvxpy -> Python MIP interface. cvxpy is only checking if cbc can be imported from cylp when it says whether cbc is installed.

**Comment by rileyjmurray** [comment]
Relevant: the next release of SciPy will support the mixed-integer programming features of HiGHS. We already have an interface to HiGHS' base LP solver through SciPy, so cvxpy users should be able to access that soon.
